### PR TITLE
Research borg, and others

### PR DESCRIPTION
**Pull request by LZakida**
This PR was actually authored by Bast, but I assisted him through a lot of it and I am pushing it from my end so it can be reviewed and such.

This PR:
+Adds a research cyborg module.
+Adds a fluff hazard vest for Bast. Unlike the other vests, this vest has no fuctional changes and is supposed to instead turn the wearer's sprite into that of a gravekeeper drone (which is a type of borg in another codebase)
*re-keys the airlock south of mechanic so the tags say aft solar instead of robotics solar or whatever it said. This was done due to a fail that occurred during the construction of the mining podbay, but logically should be done regardless seeing as the solar in question is not anywhere near where the tags suggested it was.
+puts a podbay into the mining station on z5.
+added quite a few things the above depend on, such as new sprites and cyborg items (ie grippers, swipe card)

Current issues as of initial commit:
1) The vest, when worn, only mashes the drone sprite into the existing sprite instead of making only the drone visible. I haven't really attempted to get this working yet.
2) The research borg was ported from Bay and currently has either all the same items or equivalents in our code, with the exception of one: The portable destructive analyzer. The borg is selectable and can function with the other modules it has, but can't use said porto-melter.
3) the airlock got buggered at some point and, while logically fixed, has not actually been tested to see if it works again (whee I suck!)

### CODE REVIEW

**Comment by LZakida** [comment]
fuck
this is what the first issue mentioned looks like, btw

**Comment by SheepFighter** [comment]
The RD Borg code looks fine, at a glance. The vest is going to need some overlay shenanigans to get it working. It's obviously possible though, as evidenced by things like hallucination code or the chameleon projector.

**Comment by taukausanake** [comment]
Look at the cardboard borg sprite code. With the suit and helmet on a cardboard human looks like a cyborg to other borgs but not to humans. Maybe copy that and modify it a bit

**Comment by LZakida** [comment]
I would like to mention something kinda important while I'm thinking of it.

This PR will be considered complete and mergeable when (and unless I'm persuaded otherwise given it spans more than one feature) all issues raised are addressed and dealt with. I would find it acceptable to merge if the portable D. Analyzer is not yet ported, but everything else needs to be fixed. 

As of this comment, this means:
1) The vest needs to display the sprite properly
2) The portable DA needs to work (again, this one is optional I guess)
3) The aft solar airlock and mining podbay airlock need to be fixed (I think I tested them and confirmed they do not work)
4) The Insekt sprite the research borg uses needs to be imported into robots.dmi

**Comment by SheepFighter** [comment]
For what it's worth, the vest wouldn't suffer from being in a separate PR. WIP features are all very well, but it doesn't even have any preliminary code, beyond a basic class-definition.

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 1]
```diff
@@ -271,7 +280,7 @@ var/list/robot_verbs_default = list(
 /mob/living/silicon/robot/proc/pick_module()
 	if(module)
 		return
-	var/list/modules = list("Standard", "Engineering", "Medical", "Miner", "Janitor", "Service")
+	var/list/modules = list("Standard", "Engineering", "Medical", "Miner", "Janitor", "Service")  //add     >, "Research"<     when it's ready
```
Comment (taukausanake) [inline code/modules/mob/living/silicon/robot/robot.dm]: So was the research module added?

Comment (LZakida) [reply code/modules/mob/living/silicon/robot/robot.dm]: Are you fucking with me right now

Comment (taukausanake) [reply code/modules/mob/living/silicon/robot/robot.dm]: Hey, I didn't touch this code. It's not on me

Comment (LZakida) [reply code/modules/mob/living/silicon/robot/robot.dm]: So did not one single one of my more than one multiple of my commit messages go through?

Comment (taukausanake) [reply code/modules/mob/living/silicon/robot/robot.dm]: No, I'm sure they all went through. It's just that no one pays attention to them or code reviews

**Pull request merged**